File: LeetCode/Easy/0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        int n_1 = 1;
        int n_2 = 0;
        int n_3 = 0;
        int num = 2;

        if (n < 2) {
            return n;
        }

        while (num <= n) {
            n_3 = n_1;
            n_1 += n_2;
            n_2 = n_3;
            num ++;
        }
        return n_1;
    }
}

File: LeetCode/Easy/0509-fibonacci-number/0509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        hashmap = {}
        def climb(n):
            if n <= 1:
                return n

            if n in hashmap:
                return hashmap[n]

            hashmap[n] = self.fib(n-1) + self.fib(n-2)
            return hashmap[n]

        return climb(n)


        